File: scala-train/src/main/scala/com/tanknavy/implicit_conversion/TestClass.scala
package com.tanknavy.implicit_conversion

/**
 * Author: Alex Cheng 9/14/2020 12:11 PM
 */

object TestClass {
  def main(args: Array[String]): Unit = {
    val t1 = new T1
    //t1.add()
    val t2 = T1.apply()
    //t2.add()
  }
  
  class T1{
    def add(): Unit ={
      println("class method")
    }
  }
  
  object T1{ //伴生对象？
    def apply(): T1 = {
      println("object apply method..")
      new T1()
    }
  }
}

File: scala-train/src/test/java/TestClass.java
import org.joda.time.LocalDate;

/**
 * Author: Alex Cheng 1/27/2021 3:12 PM
 */


public class TestClass {

    public static void main(String[] args) {
        String date = "2020-09-29";
        LocalDate localDate = LocalDate.parse(date);
        System.out.println(localDate);

        String date2 = localDate.toString("yyyy-MM-dd");
        System.out.println(date2);

        String date3 = localDate.toString("MM/dd/yyyy");
        System.out.println(date3);
    }


}
